[Identity] OBO sample improvement (#18330)
Azure/azure-sdk-for-js: sdk/identity/identity/samples/AzureIdentityExamples.md

Two accounts participate in the OBO flow:

- A user, which aims to obtain a special access level. Typically, the `AuthorizationCodeCredential` would be used. We'll call this identity the **User Account**.
- An app registration, which will act as the provider of the special access level. We'll call this identity the **Target App Registration**.

Both accounts must belong to the same Azure AD tenant.

While other credentials authenticate requesting access to a set of resources, the OBO flow requires the user token to have access specifically to the scope of the Azure AD app that will delegate its access to the users. For this authentication flow to work, the **Target App Registration** must be configured with a custom scope. To create a scope through the Azure portal:

1. Select **Active Directory** > **App registrations**.
2. Go to the app you want to authenticate against.
3. On the left menu, select **Expose an API** > **Add a scope**.

> Besides that, the **Target App Registration** must have admin consent. It can be granted as follows:

Reviewer: ```suggestion
The **Target App Registration** must also have admin consent, which can be granted as follows:
```
